### ISSUE TITLE
FRAGMOS - cardinality of observationsDates : to be (0..1) instead of (1..1)

### ISSUE THREAD

**Issue by JBZ-Fragmos** [finos/common-domain-model#2140]
when at least calculationPeriodDates exists, there is not necessarily a business need to display as well observationDates, which in most cases will lead to redundant kind of information (because these 2 objects both represent periods, etc.)

not to say either should be removed, because in some cases both may exist / be populated for the same Trade

here it is today, for clarity
![image](https://user-images.githubusercontent.com/92978493/236514685-33200154-a602-4c21-b499-a52b1e96c23c.png)

not to say either should be removed, because in some cases both may exist / be populated for the same Trade

all in all, at least 1 or both must exist, and we may assume that observationDates

accordingly, **PROPOSAL** is to :
**1 - updated cardinality of observationDates : to become (0..1)**
**2 - and to ensure that at least observationDates exists if calculationPeriodDates is absent, thus adding this " condition : if observationDates is absent then calculationPeriodDates exists "**

**Comment by tomhealey-icma** [comment]
I agree and while changes are being made can we remove FxSpotRateSource, its too market specific for a general set of ObservationTerms.

**Comment by JBZ-Fragmos** [comment]
agree as well with changing FxSpotRateSource into more generic type name... 
I guess we shall simply rename it exactly the same
so "InformationSource" woud be the new name of "FxSpotRateSource", right ?

**Comment by JBZ-Fragmos** [comment]
as far as I'm concerned (n behalf of FragmosChain, of course) it's been a time I've been pushing for refactoring ObservationTerms in more generic way... so I'm happy if you are with me on this one...

more to the point, my proposal is that ObservationTerms should conceptually anwer to these 3 questions : 

- what (do we observe) ? that should answer you question 1. definitly i agree : an observable is not only a price per se (please see my detailed answer below)
- when (do we observe) ? see detailed answer below, do not agree ObservationPeriod shall replace current object because it is already included in it as far as i can see (cf. PeriodicDates inside)
- how (do we observe) ? should answer to your question 2. and in 1 word i agree with you we need a calculation method here

- what == CDM object : Observable, already there, though would recommand refactoring as proposed in my **Issue n°#2106**
- when = CDM object : ObservableDates (i would not say as you to change label to ObservationPeriod because "Periods" is a sub-type of dates among other possible types, so would prefer satying more generic... then you can define Periods with the PeriodicDates that currently exist as a possible sub-type inside the ObservationDates... also do not see obvious Business Case where you use "numberOfObservationPeriods" : have you any example in mind ?)
- how == CDM objects : we need to gather existing CDM objects for instance InformationSource, Rounding currently present as distinct items in ObservationTerms, but also other objects, notably DeterminationMethodEnum that really missing here in my opinion... as per your question 2.

all in all fpr your info as part of the modelling work i've done for Structuring Group i've faced already such need to refactor observable terms... hereunder my proposal : happy to provide more details if you would like so...

![image](https://github.com/finos/common-domain-model/assets/92978493/71f00b5a-d83f-42e5-b34d-eb8d0244c198)

for clarity, the "aggregationFeature" does not exist yet in CDM it is part of my workspace, that is to provide witn generic aggregation rules either per date when multiple observable exist (basket) or per underlier when multiple dates exists... this is an object i've built fopr the Structured Modelling, but could be used by default in any case in my opinion... this object would notably answer to your question 2.

![image](https://github.com/finos/common-domain-model/assets/92978493/43c53f9a-ac2c-4e89-af7e-e62fcf1a997e)

**Comment by JBZ-Fragmos** [comment]
all in all hereabove discussion is interesting for sure, but i beleive it is going a bot far from original proposal that is actually a straight and easy proposal - i'm giving it again for clarity, 

- kindly review for **approval** 
- and I will send related **PR** : 

**PROPOSAL** is to :
**1 - updated cardinality of observationDates : to become (0..1)
2 - and to ensure that at least observationDates exists if calculationPeriodDates is absent, thus adding this " condition : if observationDates is absent then calculationPeriodDates exists "**
